### ISSUE TITLE
Wrong language selected

### ISSUE THREAD

**Issue by dimaqq** [sozialhelden/wheelmap-frontend#262]
macos, Chrome, the browser is configured so:

```
window.navigator.language
"en-GB"
window.navigator.languages
["en-GB", "en", "fi"]
```

the request header contains `accept-language: en-GB,en;q=0.9,fi;q=0.8`

And Finnish is selected by the frontend 🤷

**Comment by opyh** [comment]
Thanks for the report! Reproduced - we have no British translation (only `en-US`), the frontend prefers the exact match `fi` over the replaced fallback existing locale `en-US` inferred from 'en'. If you remove `fi`, `en-US` is selected as best guess instead. I see why you probably expected `en-US` to be preferred.

I'd hardwire 'en' to 'en-US' as default in the next deployment.

Other considered options:

1) Create and maintain a British translation
2) Append an arbitrary country code to match an existing translation if no existing translation is found and prefer the match. This would cause problems for languages that have stronger differences between country codes though, e.g. Chinese.

Other ideas? :)

**Comment by dimaqq** [comment]
Based on an informal poll at my company (international crowd), English is preferred over "wrong" Chinese with caveats.
I don't know if this can be extrapolated to monolinguists.

**Comment by opyh** [comment]
Thanks for collecting this information! :)

While working on this I tested and compared several language matching implementations (builtin Java, Go, Rust, several JS libraries, http://cldr.unicode.org/index/charts, https://tools.ietf.org/html/rfc4647), but found that none of them accurately worked with the variety of Chinese (legacy) subtags, so I added a hardcoded matcher to work around this:

https://github.com/sozialhelden/wheelmap-frontend/blob/dcd07a2b0471cff3c3cf370b48baa24aadcef309/src/lib/i18n.js#L74

It currently falls back to `zh-Hans` if your browser/system is set to `zh`, while Wheelmap _does_ support both Traditional and Simplified Chinese internally. From your comment it seems we shouldn't do so (even given that this will work for 90% of the users) but rather find out the visitor's country by IP, and use it for a better guess.
